MayssaraOmar/assignment3/Word search/src/test_wordSearch.cpp

```diff
@@ -0,0 +1,136 @@
+#include "test_wordSearch.h"
+#include "word search.h"
+#include "test.h"
+
+dictionary test_wordSearch::dic;
+
+void test_wordSearch::setup() {
+	dic.insert("car");
+	dic.insert("card");
+	dic.insert("cart");
+	dic.insert("cat");
+	dic.insert("m");
+}
+void test_wordSearch::run_all_tests() {
```

Comment:
Similarly to above, it would make sense to run all the tests on a different instance to avoid interactions between them.

Since you have a setup method you could:
* call setup() in the constructor
* call setup() explicitly for each test

For the latter on, you could use:

```c++
test_wordSearch& test_WordSearch::setup() {
  ...
  return *this;
}

...


void test_wordSearch::run_all_tests() {
  new test_wordSearch().setup().empty_grid();
  new test_wordSearch().setup().one_cell_grid_with_match_in_dictionary();
  ...
}
```

Does this make sense?

Reply:
This is what I understood:

1- If I call setup() in the constructor, then all tests still share the same dictionary and there are interactions between tests(unless I'm making a new instance for each test). 

2- I don't need to make  a different instance for each test if each test is completely independent (has its own dictionary), for example calling  setup(dictionary*) explicitly for each test with its own dictionary.

I think method 2 is simpler as I don't have the responsibility of freeing the memory I allocated for each test.